cmd/earthly/app/before.go

```diff
@@ -23,69 +23,74 @@ import (
 )
 
 func (app *EarthlyApp) before(cliCtx *cli.Context) error {
-	if app.BaseCLI.Flags().EnableProfiler {
+	flags := app.BaseCLI.Flags()
+
+	if flags.EnableProfiler {
 		go profhandler()
 	}
 
-	if app.BaseCLI.Flags().InstallationName != "" {
+	if flags.InstallationName != "" {
 		if !cliCtx.IsSet("config") {
-			app.BaseCLI.Flags().ConfigPath = defaultConfigPath(app.BaseCLI.Flags().InstallationName)
+			flags.ConfigPath = defaultConfigPath(flags.InstallationName)
 		}
 		if !cliCtx.IsSet("buildkit-container-name") {
-			app.BaseCLI.Flags().ContainerName = fmt.Sprintf("%s-buildkitd", app.BaseCLI.Flags().InstallationName)
+			flags.ContainerName = fmt.Sprintf("%s-buildkitd", flags.InstallationName)
 		}
 		if !cliCtx.IsSet("buildkit-volume-name") {
-			app.BaseCLI.Flags().BuildkitdSettings.VolumeName = fmt.Sprintf("%s-cache", app.BaseCLI.Flags().InstallationName)
+			flags.BuildkitdSettings.VolumeName = fmt.Sprintf("%s-cache", flags.InstallationName)
 		}
 	}
-	if app.BaseCLI.Flags().Debug {
+	if flags.Debug {
 		app.BaseCLI.SetConsole(app.BaseCLI.Console().WithLogLevel(conslogging.Debug))
-	} else if app.BaseCLI.Flags().Verbose {
+	} else if flags.Verbose {
 		app.BaseCLI.SetConsole(app.BaseCLI.Console().WithLogLevel(conslogging.Verbose))
 	}
-	if app.BaseCLI.Flags().LogstreamUpload {
-		app.BaseCLI.Flags().Logstream = true
+
+	app.BaseCLI.SetConsole(app.BaseCLI.Console().WithPrefixWriter(app.BaseCLI.Logbus().Run().Generic()))
+	if flags.BuildID == "" {
+		flags.BuildID = uuid.NewString()
 	}
-	if app.BaseCLI.Flags().Logstream {
-		app.BaseCLI.SetConsole(app.BaseCLI.Console().WithPrefixWriter(app.BaseCLI.Logbus().Run().Generic()))
-		if app.BaseCLI.Flags().BuildID == "" {
-			app.BaseCLI.Flags().BuildID = uuid.NewString()
-		}
-		var execStatsTracker *execstatssummary.Tracker
-		if app.BaseCLI.Flags().ExecStatsSummary != "" {
-			execStatsTracker = execstatssummary.NewTracker(app.BaseCLI.Flags().ExecStatsSummary)
-		}
-		disableOngoingUpdates := !app.BaseCLI.Flags().Logstream || app.BaseCLI.Flags().InteractiveDebugging
-		forceColor := envutil.IsTrue("FORCE_COLOR")
-		noColor := envutil.IsTrue("NO_COLOR")
-		var err error
-		newSetup, err := logbussetup.New(
-			cliCtx.Context, app.BaseCLI.Logbus(), app.BaseCLI.Flags().Debug, app.BaseCLI.Flags().Verbose, app.BaseCLI.Flags().DisplayExecStats, forceColor, noColor,
-			disableOngoingUpdates, app.BaseCLI.Flags().LogstreamDebugFile, app.BaseCLI.Flags().BuildID, execStatsTracker, app.BaseCLI.Flags().GithubAnnotations)
-		app.BaseCLI.SetLogbusSetup(newSetup)
-		if err != nil {
-			return errors.Wrap(err, "logbus setup")
-		}
+	var execStatsTracker *execstatssummary.Tracker
+	if flags.ExecStatsSummary != "" {
+		execStatsTracker = execstatssummary.NewTracker(flags.ExecStatsSummary)
+	}
+	newSetup, err := logbussetup.New(
```

Comment:
```suggestion
	theOnlySetup, err := logbussetup.New(
```

or maybe just `setup` :laughing:

Reply:
Lol. I don't if that name means "the setup using the new approach" or just "the newly created setup variable". I'll clarify it, nonetheless.

Reply:
yeah `setup` seems like a bad name; I too don't really know what it means at first glance.

Reply:
Sorry, I meant the existing name `newSetup`.